docs/tutorials/developer-journey/level-2/2.1-storage-persistence.md

```diff
@@ -0,0 +1,189 @@
+# 2.1: Canister upgrades, storage, and persistence 
+
+## Overview
+
+The Internet Computer handles long-term data storage using a feature known as **stable memory**. **Stable memory** is a unique feature of the Internet Computer that defines a data store separate from the canister's regular Wasm memory data store, which is known as **heap storage**. A canister's heap storage is not long-term storage and does not persist across canister upgrades; canister data and state stored in heap storage is removed when a canister is stopped, reinstalled, or upgraded. Stable memory is important since it persists across canister upgrades, has a much larger storage capacity than heap storage, and is very beneficial for vertically scaling a dapp. 
+
+To use stable memory requires anticipating and indicating which canister data you want to be retained after a canister upgrade. For some canisters, this data might be all of the canister's data, while for others it may be only certain parts or none. By default, stable memory starts empty, and can hold up to 32GB as long as the subnet the canister is running on has the available space. If a canister uses more stable memory than 32GB, the canister will trap and become unrecoverable. 
```

Comment:
```suggestion
To use stable memory requires anticipating and indicating which canister data you want to be retained after a canister upgrade. For some canisters, this data might be all of the canister's data, while for others it may be only certain parts or none. By default, stable memory starts empty, and can hold up to 96GiB as long as the subnet the canister is running on has the available space. If a canister uses more stable memory than 96GiB, the canister will trap and become unrecoverable. 
```